source/remediation_runbooks/scripts/test/test_replacecodebuildcleartextcredentials.py

```diff
@@ -9,30 +9,32 @@
 
 import ReplaceCodeBuildClearTextCredentials as remediation
 
-my_session = boto3.session.Session()
-my_region = my_session.region_name
-
-BOTO_CONFIG = Config(
-    retries = {
-        'mode': 'standard'
-    },
-    region_name = my_region
-)
+def get_region() -> str:
+    my_session = boto3.session.Session()
+    return my_session.region_name
+
+def get_config() -> Config:
+    return Config(
+        retries = {
+            'mode': 'standard'
+        },
+        region_name = get_region()
+    )
```

Comment:
This has some duplication across tests. Maybe extract into some sort of BotoMock helper class?

Reply:
That is out of scope for this change. I am trying to get the unit test workflow to work, which allows us to trust that when a workflow fails it means something. That is extremely important. Code duplication is far less important.

I also would not like to invest in reducing duplication of this code because I think it's actually wrong. For tests there's no need to specify a config. We should never be retrying in unit tests. The way regions are treated in tests is also pretty inconsistent. In this case, I think it's totally unnecessary to add it to the config. We're fetching from the default region of the session, which in the case of tests will be set by mock env vars, as it should be. Then we're setting the region of the config to that value. Without that value, the region of clients created from the default session would be fetched from the same exact env var.

Reply:
I would probably argue that encapsulating the bad code is still better if only because we can do a find_usages report against the iffy method call when we come to fix it later. but it is indeed a minor thing.